gateway/src/main/java/com/mx/path/gateway/connect/filter/TracingFilter.java

```diff
@@ -20,27 +20,30 @@
 public class TracingFilter extends RequestFilterBase {
   @Override
   public final void execute(Request request, Response response) {
-    Span span = CustomTracer.startSpan(request.getTraceKey());
-    if (span == null) {
+    Tracer tracer = GlobalTracer.get();
+
+    if (tracer == null) {
       next(request, response);
       return;
     }
 
-    try (Scope scope = CustomTracer.activateSpan(span)) {
-      try {
-        Map<String, String> traceHeaders = new LinkedHashMap<>();
-
-        CustomTracer.injectSpanContextIntoHeaders(traceHeaders);
-        MultiValueMap<String, String> requestHeaders = new MultiValueMap<>(request.getHeaders());
-        traceHeaders.forEach(requestHeaders::add);
-
-        request.setTraceId(span.context().toTraceId());
-        request.setTraceSpanId(span.context().toSpanId());
-        next(request, response);
-      } catch (Exception e) {
-        span.log(e.getMessage());
-        throw e;
-      }
+    Span span = tracer.buildSpan(request.getTraceKey())
+        .start();
+    // Note: The scope in the try with resource block below
```

Comment:
Thank you for being explicit here so we don't run into accidental memory leaks down the road